pkg/bmc/virtualmedia_validation.go

```diff
@@ -0,0 +1,94 @@
+package bmc
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/stmcginnis/gofish"
+	"github.com/stmcginnis/gofish/redfish"
+)
+
+// contains tells whether the array of redfish.VirtualMediaType
+// has the element redfish.VirtualMediaType.
+func contains(array []redfish.VirtualMediaType, element redfish.VirtualMediaType) bool {
+	for _, n := range array {
+		if element == n {
+			return true
+		}
+	}
+	return false
+}
+
+// checkVirtualMedia tells if all virtual media of a given Manager
+// has the correct type and actions to work with virtual media boot.
+func checkVirtualMedia(m *redfish.Manager) error {
+	vmediaList, err := m.VirtualMedia()
+	if err != nil {
+		return err
+
+	}
+	if len(vmediaList) == 0 {
+		return fmt.Errorf("no virtual media found for manager %s", m.ID)
+	}
+
+	var vmediaError error
+	for _, vmedia := range vmediaList {
+		if !contains(vmedia.MediaTypes, redfish.CDMediaType) {
+			vmediaError = fmt.Errorf("virtual media %s from manager %s does not contain "+
+				"CD MediaType", vmedia.ID, m.ID)
+			continue
+		}
+		if !vmedia.SupportsMediaEject || !vmedia.SupportsMediaInsert {
+			vmediaError = fmt.Errorf("virtual media %s from manager %s does not support "+
+				"at least one of following Actions: InsertMedia, EjectMedia", vmedia.ID, m.ID)
+			continue
+		}
+		return nil
+	}
+	return vmediaError
+}
+
+// ValidateVirtualMedia returns nil when the BMC has support
+// for VirtualMedia and error when the BMC does not have support.
+func ValidateVirtualMedia(address string, user string, password string, insecure bool, systemURI string, httpClient *http.Client) error {
```

Comment:
very nit: a weird ordering of arguments, I feel like systemURI should come right after address/user/password.

Reply:
Sure